### PR TITLE
small bugfixes

### PR DESCRIPTION
**Pull request by fscherf**
This PR contains two little bugfixes: One for the Javascript client, and one regarding routing problems

### CODE REVIEW

**Review by maratori — APPROVED** [review]
LGTM

File: lona/routing.py [hunk 1]
```diff
@@ -35,27 +35,26 @@ def __init__(self, raw_pattern, view, name='', interactive=True,
...
         else:
-            if self.raw_pattern.endswith(OPTIONAL_TRAILING_SLASH_PATTERN):
+            raw_pattern = self.raw_pattern
+
+            if raw_pattern.endswith(OPTIONAL_TRAILING_SLASH_PATTERN):
                 self.optional_trailing_slash = True
 
-            groups = ABSTRACT_ROUTE_RE.findall(self.raw_pattern)
+                raw_pattern = \
+                    raw_pattern[:len(OPTIONAL_TRAILING_SLASH_PATTERN)*-1]
```
Comment (laundmo) [inline lona/routing.py:44]: `*-1` seems a bit odd here, why not `- len(OPTIONAL_TRAILING_SLASH_PATTERN)`

**Review by laundmo — COMMENTED** [review]
small oddity, but otherwise LGTM

File: lona/routing.py (hunk 1 above)
Comment (fscherf) [reply lona/routing.py]: Ack! Fixed

**Review by laundmo — APPROVED** [review]

**Comment by fscherf** [comment]
@maratori, @laundmo: Thanks for reviewing!

**Pull request merged**